src/printer.js

```diff
@@ -548,29 +548,46 @@ function genericPrintNoParens(path, options, print) {
         "body"
       );
 
-      // If there are no contents, return a simple block
-      if (!getFirstString(naked)) {
+      const hasDirectives = n.directives && n.directives.length > 0;
+      const hasContent = getFirstString(naked);
+      const hasInnerComments = n.innerComments && n.innerComments.length > 0;
+
+      if (!hasInnerComments && !hasDirectives && !hasContent) {
         return "{}";
       }
 
       parts.push("{");
 
-      // Babel 6
-      if (n.directives) {
+      // Babylon 6
+      if (hasInnerComments) {
+        path.each(function(innerComment) {
+          parts.push(
+            indent(
+              options.tabWidth,
+              concat([ hardline, print(innerComment) ])
+            )
+          );
+        }, "innerComments");
```

Comment:
A little more explanation: this is weird to me because this is a property on a node that we didn't create in our "attach comments" pass. If you look in the `attach` function in comments.js you'll see how we manually attach comments throughout the AST and create our own properties. This comes straight from recast and I think exists because we can be a lot more accurate about which comments leading/trailing. That logic seems to call this a "dangling" comment and I'd like to pull comments from a single source of truth: either our custom properties or the default AST properties. Let's not mix them.